lib/ts/controllers/s-navigation-tablist.ts

```diff
@@ -0,0 +1,121 @@
+namespace Stacks {
+    export class TabListController extends Stacks.StacksController {
+
+        private boundSelectTab: any; // (event: MouseEvent) => void;
+        private boundHandleKeydown: any // (event: KeyboardEvent) => void;
+
+        connect() {
+            super.connect();
+
+            this.boundSelectTab = this.selectTab.bind(this);
+            this.boundHandleKeydown = this.handleKeydown.bind(this);
+
+            for (let tab of this.tabTargets) {
+                tab.addEventListener("click", this.boundSelectTab);
+                tab.addEventListener("keydown", this.boundHandleKeydown);
+            }
+        }
+
+        disconnect() {
+            super.disconnect();
+
+            for (let tab of this.tabTargets) {
+                tab.removeEventListener("click", this.boundSelectTab);
+                tab.removeEventListener("keydown", this.boundHandleKeydown);
+            }
+        }
+
+        /**
+         * Gets all tabs within the controller.
+         */
+        get tabTargets() {
+            return <HTMLElement[]>Array.from(this.element.querySelectorAll("[role=tab]"));
+        }
+
+        /**
+         * Handles click events on individual tabs, causing them to be selected.
+         */
+        selectTab(event: MouseEvent) {
+            this.switchToTab(<HTMLElement>event.currentTarget);
+
+            // We need to be very agressive about selecting the current tab on click.
+            // Mouse click will not always achieve this and failing to do so will prevent
+            // keyboard navigation from working.
+            this.selectedTab?.focus();
```

Comment:
Oh I just hate it though ☹️ Is there any way we can avoid a hard `focus` on these? I really don't want there to be the regression of killing `:focus` styling. I'd love to rely on `:focus-visible` which is why I went on that tangent.

Reply:
Maybe we can?  The issue was trying to avoid was users clicking on a tab and then being unable to keyboard navigate it, something that affects Firefox and Safari but not Chrome.  Looking at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_Role, their example has the exact same issue so maybe it's fine to only give keyboard controls to users who keyboard into it.

Reply:
I think that'd be a worthy tradeoff. The thing I want to avoid is focus states for simple clicking, but of course, I want focus states for those keyboard folks.

Reply:
This already has the `:focus-visible` styling in it. The problem I'm seeing with calling `.focus()` is that it adds focus styling on the element regardless of the browser's default behavior. We really only want focus styling if the user used their keyboard.